bricks-maven-plugin/src/main/java/com/github/ingogriebsch/bricks/maven/plugin/analyzer/MavenAnalyzer.java

```diff
@@ -0,0 +1,64 @@
+/*-
+ * #%L
+ * Bricks Maven Plugin
+ * %%
+ * Copyright (C) 2018 - 2019 Ingo Griebsch
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.github.ingogriebsch.bricks.maven.plugin.analyzer;
+
+import java.util.Comparator;
+
+import com.github.ingogriebsch.bricks.model.Component;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+
+import lombok.NonNull;
+
+/**
+ * Main entrypoint for an analyzer to do its magic and contribute generated
+ * insight to the Component given.
+ * 
+ * To do it "right" rather than using ordinals, we could build a dependency
+ * graph of analyzers and even run them in parallel. Imho not worth it right
+ * now, because plugin is only supposed to run on package phase.
```

Comment:
package phase? Sounds not legit if thinking about the phases. How about generate-resources or -sources or process-resources or -sources?